### PR TITLE
Enhance documentation

### PR DESCRIPTION
**Pull request by chfast**
- Updates the EVMC module description and moves it to a markdown file.
- Some config tweaks for Doxygen.
- Documentation for macros from utils.h.

### CODE REVIEW

**Comment by chfast** [comment]
See https://ethereum.github.io/evmc/.

File: README.md [hunk 1]
```diff
@@ -49,7 +49,7 @@ Licensed under the [MIT License](LICENSE.md).
 [@axic]: https://github.com/axic
 [@chfast]: https://github.com/chfast
 [documentation]: https://ethereum.github.io/evmc
-[eWASM]: https://github.com/ewasm/design#ethereum-flavored-webassembly-ewasm-design
+[Ewasm]: https://github.com/ewasm/design#ethereum-flavored-webassembly-ewasm-design
```
Comment (gumb0) [inline README.md]: This anchor changed, I think now it's https://github.com/ewasm/design#what-is-ethereum-flavored-webassembly-ewasm

File: README.md [hunk 2]
```diff
@@ -6,7 +6,7 @@
 > Ethereum Client-VM Connector API
 
 The EVMC is the low-level ABI between Ethereum Virtual Machines (EVMs) and 
-Ethereum Clients. On the EVM-side it supports classic EVM1 and [eWASM].
+Ethereum Clients. On the EVM-side it supports classic EVM1 and [Ewasm].
```
Comment (gumb0) [inline README.md]: I think "EVM side" should be without dash

File: docs/EVMC.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+
+## Modules
+
+- [EVMC](@ref EVMC) 
+   – the main component that defines API for VMs and Clients (Hosts).
+- [EVMC Loader](@ref loader)
+   – the library for loading VMs implemented as Dynamically Loaded Libraries (DLLs, shared objects). 
+- [EVMC Helpers](@ref helpers)
+   – a collection of utility function for easier integration with EVMC.
```
Comment (gumb0) [inline docs/EVMC.md]: functions?

File: docs/EVMC.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+- [EVM Instructions](@ref instructions) 
+   – the library with collection of metrics for EVM1 instruction set.
+
+
+[Ewasm]: https://github.com/ewasm/design#ethereum-flavored-webassembly-ewasm-design
```
Comment (gumb0) [inline docs/EVMC.md]: https://github.com/ewasm/design#what-is-ethereum-flavored-webassembly-ewasm

Comment (axic) [reply docs/EVMC.md]: Why does it need to link to the heading?

File: README.md (hunk 2 above)
Comment (axic) [reply README.md]: I don't think there's an agreement on the name, can you just have the wasm stuff in a separate PR?

File: docs/EVMC.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+- Controls the call depth, including the exceptional termination of execution
+  in case the maximum depth is reached.
+  
+  
+### Host
+
+- Provides access to State.
+- Creates new accounts (with code being a result of VM execution).
+- Handles refunds entirely.
+- Handles calls to precompiled contracts.
```
Comment (axic) [inline docs/EVMC.md]: What does that exactly mean?

File: docs/EVMC.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  counter.
+- Controls the call depth, including the exceptional termination of execution
+  in case the maximum depth is reached.
+  
+  
+### Host
+
+- Provides access to State.
+- Creates new accounts (with code being a result of VM execution).
+- Handles refunds entirely.
```
Comment (axic) [inline docs/EVMC.md]: Actually it may make sense moving that to the VM at some point.

**Comment by axic** [comment]
Okay apar from the ewasm naming question.

**Pull request merged**